spec/controllers/kudos_controller_spec.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe KudosController do
+  include LoginMacros
+  include RedirectExpectationHelper
+
+  describe "create" do
+    context "when work is public" do
+      let(:work) { create(:posted_work) }
+
+      context "when kudos are given from work" do
+        it "redirects to referer with a notice" do
+          referer = work_path(work)
+          request.headers["HTTP_REFERER"] = referer
```

Comment:
This is the (recently) recommended way of setting request headers in controller specs:

    https://github.com/rspec/rspec-rails/pull/2229

It looks marginally cleaner than:

https://github.com/otwcode/otwarchive/blob/540f520aef15d32d57a85e5c9751a2924cea7fa5/spec/controllers/inbox_controller_spec.rb#L136-L142